tests/api2/test_catalog_sync.py

```diff
@@ -37,12 +38,30 @@ def kubernetes_pool():
             'repository': 'https://github.com/truenas/charts.git',
             'branch': 'test'
         }):
+            timeout = 150
+            while True:
+                time.sleep(5)
+                kube_system_pods = call(
+                    'k8s.pod.query', [
+                        ['metadata.namespace', '=', 'kube-system']
+                    ], {'select': ['metadata.name', 'status.phase']}
+                )
+                if len([pod for pod in kube_system_pods if pod['status']['phase'] == 'Running']) >= 3 or timeout <= 0:
+                    break
```

Comment:
I don't understand much what's going on, but do we yield the same result if there are 3 pods running or there are less but a timeout occurred? Shouldn't we stop the tests if a timeout occurs?